lib/meta-storage.js

```diff
@@ -0,0 +1,41 @@
+'use strict';
+
+const assert = require('assert');
+
+module.exports = class MetaStorage {
+    constructor(sink) {
+        assert(sink, 'MetaStorage needs a sink');
+        this.sink = sink;
+    }
+
+    getKey(key) {
+        return `/meta/${key}.json`;
+    }
+
+    async set(fileName, fileContent) {
+        assert(fileName, '"key" required');
+        assert(fileContent, '"value" required');
+        return this.sink.set(
+            this.getKey(fileName),
+            JSON.stringify(fileContent)
+        );
+    }
+
+    async get(fileName) {
+        const result = await this.sink.get(this.getKey(fileName));
+        if (result) {
+            try {
+                return JSON.parse(result);
+            } catch (e) {
+                throw new Error(
+                    `Failed parsing payload from key "${fileName}"`
```

Comment:
What is "payload"?

Reply:
Please keep the original error as well, so it's possible to see why it failed

Reply:
Lets improve in new pr.

Reply:
why?

Reply:
I have upcoming changes here soon, will add this.